xdsl/irdl/constraints.py

```diff
@@ -55,6 +56,15 @@ def update(self, other: ConstraintContext):
 
 _AttributeCovT = TypeVar("_AttributeCovT", bound=Attribute, covariant=True)
 
+ResolveType: TypeAlias = Attribute | Sequence[Attribute]
+
+_T = TypeVar("_T")
+
+
+class Resolver(Generic[_T], abc.ABC):
```

Comment:
Isn't this an `AttrInference` or something?
I feel using the `Inference` word makes more sense than `Resolve` here

Reply:
I was considering something along the lines of `Extractor`. The idea is that it describes how to get the value of a variable from the input `_T` (which could be an attribute, sequence of attributes, ParsingState etc)

Reply:
Hmm, I see what you mean, I think `Extractor` makes more sense then indeed! Or `VarExtractor` to be more precise?